Source/Dafny/Dafny.atg

```diff
@@ -3820,17 +3837,45 @@ ParensExpression<out Expression e>
      var args = new List<ActualBinding>();
   .)
   "("                                        (. x = t; .)
-  [ ActualBindings<args> ]
+  [ TupleArgs<args> ]
   ")"
-  (. if (args.Count == 1) {
+  (. if (args.Count == 1 && !args[0].IsGhost) {
        if (args[0].FormalParameterName != null) {
          SemErr(args[0].FormalParameterName, "binding not allowed in parenthesized expression");
        }
        e = new ParensExpression(x, args[0].Actual);
      } else {
+       // Compute the actual position of ghost arguments
+       // TODO(fpoli): Is this the correct place to do this? Should this be moved to the resolver?
```

Comment:
I have a question here.

Reply:
Sorting is not needed. The argument name can be parsed and used as an index (if in range).